config/internal/mariadb/pvc.yaml.tmpl

```diff
@@ -9,6 +9,7 @@ metadata:
 spec:
   accessModes:
     - ReadWriteOnce
+  storageClassName: {{.MariaDB.StorageClassName}}
```

Comment:
Have we tested to ensure if storageClassName was set to `""` instead of an actual storage class name, as shown in custom config dspa.yaml below, this still works?  It feels like it shouldn't (and therefore we should have logic to only add the item if an explicit value is provided) but perhaps kube/openshift has magic for it

Reply:
I'm actually seeing an error when I don't specify storageclass: 

```
2023-12-07T19:38:59Z	ERROR	Reconciler error	{"controller": "datasciencepipelinesapplication", "controllerGroup": "datasciencepipelinesapplications.opendatahub.io", "controllerKind": "DataSciencePipelinesApplication", "DataSciencePipelinesApplication": {"name":"sample","namespace":"dspa"}, "namespace": "dspa", "name": "sample", "reconcileID": "46523bbf-59c5-4bf8-9819-1c1c0f9a2768", "error": "PersistentVolumeClaim \"mariadb-sample\" is invalid: spec: Forbidden: spec is immutable after creation except resources.requests for bound claims\n  core.PersistentVolumeClaimSpec{\n  \t... // 2 identical fields\n  \tResources:        {Requests: {s\"storage\": {i: {...}, s: \"10Gi\", Format: \"BinarySI\"}}},\n  \tVolumeName:       \"pvc-1bee7a8f-94b0-48f3-b036-c9f87536f73c\",\n- \tStorageClassName: &\"standard-csi\",\n+ \tStorageClassName: nil,\n  \tVolumeMode:       &\"Filesystem\",\n  \tDataSource:       nil,\n  \tDataSourceRef:    nil,\n  }\n"}
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler
	/opt/app-root/src/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.13.0/pkg/internal/controller/controller.go:326
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem
	/opt/app-root/src/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.13.0/pkg/internal/controller/controller.go:273
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2
	/opt/app-root/src/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.13.0/pkg/internal/controller/controller.go:234

```

@VaniHaripriya can you take a look at the operator logs when you try to deploy a simple dspa without any storageclassname field set?

Reply:
+1 to what @gmfrasca and @HumairAK mentioned, when the storageClassName is set to "", this doesn't work.